server/app/genbank.py

```diff
@@ -102,3 +112,96 @@ def convert_gbk_to_vector(genbank_file, level: VectorLevel) -> GenbankData:
         features=features,
         references=references,
     )
+
+
+# Converts a model.Vector to a Genbank output
+def convert_LevelN_to_genbank(
+    vector: model.Vector, database: Session = Depends(deps.get_db)
+) -> str:
+    vector_record = SeqIO.SeqRecord(seq=Seq(vector.sequence))
+
+    # General construct information
+    vector_record.name = (
+        "MP-G1-" + str(vector.location) + "_" + vector.name.replace(" ", "_")
+    )
+    vector_record.description = "synthetic circular DNA"
+
+    vector_record.origin = ""
+    vector_record.size = len(vector.sequence)
+
+    # Annotations
+    for annotation in vector.annotations:
+        vector_record.annotations[annotation.key] = annotation.value
+
+    vector_record.accession = []
+    vector_record.comment = ""
+    vector_record.data_file_division = ""  # E.g. 'PLN' stands for plants
+    vector_record.date = vector.date  # Is also present in annotations...
+    vector_record.keywords = []
+    vector_record.residue_type = "ds-DNA circular"
+    vector_record.organism = (
+        "synthetic DNA construct"  # To ask: Should this be bacterial_strain?
+    )
+    # To ask: molecule_type == residue_type?
+    # To ask: sequence_version == accesion version?
+    # The source of material where the sequence came from.
+    vector_record.source = []
+    vector_record.taxonomy = (
+        []
+    )  # A listing of the taxonomic classification of the organism, starting general and getting more specific.
+    vector_record.topology = ""
+
+    # References
+    vector_record.annotations["references"] = [
+        mod_Reference(authors=ref.authors, title=ref.title) for ref in vector.references
+    ]
+
+    # Features
+    feature_list = []
+    for vector_feature in vector.features:
+        feature_qualifiers = vector_feature.qualifiers
+
+        feat = Bio.SeqFeature.SeqFeature(
+            type=vector_feature.type,
+            location=Bio.SeqFeature.FeatureLocation(
+                start=vector_feature.start_pos,
+                end=vector_feature.end_pos,
+                strand=vector_feature.strand,
+            ),
+        )
+
+        for qual in feature_qualifiers:
+            feat.qualifiers[qual.key] = qual.value
+
+        # vector_record.features.append(feat)
+        feature_list.append(feat)
+
+    vector_record.features = feature_list
+
+    # Writing record content to an in-memory file and returning its content.
+    outf = io.StringIO()
+    SeqIO.write(vector_record, outf, "genbank")
+    content = outf.getvalue()
+    outf.close()
+    return content
```

Comment:
Turns out this can be done with a context manager:
```suggestion
    with io.StringIO() as outf:
        SeqIO.write(vector_record, outf, "genbank")
        return outf.getvalue()
```